MarkBind/markbind
# Create "Scroll-to-Top" button

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] New feature
This PR addresses issue #654 by adding a Scroll-to-Top button on every page rendered. 

**What is the rationale for this request?**

This would improve user experience by providing them a quick button to scroll to the top of the page. 

**What changes did you make? (Give an overview)**

When Page adds "Content Wrapper" to pageData, I introduced another step to add a Scroll-to-Top button to the end of the pageData html. 

The "Scroll-to-Top" button can be found on the **bottom-right corner** of the window. 

I have also added some dynamic features to it: 
1) the button will only appear after user scrolls down, 
2) the button will disappear when the user has fully scrolled to the top, 
3) hover causes the button to darken from opacity 40% to 70%. 

**Browser View**:
![Screen Shot 2020-08-30 at 3 55 29 PM](https://user-images.githubusercontent.com/37920115/91654163-4d113e80-ead9-11ea-9380-061a0fb2dcc6.png)

**Mobile View**:

![Screen Shot 2020-08-30 at 3 56 44 PM](https://user-images.githubusercontent.com/37920115/91654185-6a460d00-ead9-11ea-841d-6fabfc038907.png)

**Is there anything you'd like reviewers to focus on?**

1) Code quality

2) Currently, the button will appear on both **mobile view** and **browser view**. There is currently an option for users to press the "Home" button on their keyboard to go to the top of the page. 

But I still feel that there is an incentive in making sure it appears on the **browser view** as well in case user doesn't know about the "Home" keyboard button feature. 

It also provides the user some minor convenience in that they most likely would already have their hand on the mouse, so they can just click on the "Scroll-to-Top" button, whereas it is less likely that they would have their hand on the keyboard (as compared to the mouse). 

3) The button design -- whether the current button is fine or we should opt for other designs. The current design was chosen from FontAwesome, specifically "fa-arrow-circle-up". 

**Testing instructions:**

1) ```markbind serve -d``` on an appropriate Markbind project folder (e.g. the current user guide)
2) Scroll down from the top, as the scroll comes to a halt, you should see the "Scroll-to-Top" button appear
3) After clicking on the button, your screen should scroll to the top of the page **smoothly** and the button should disappear

Note: Remember to use the ```-d``` option when you serve the project (```markbind serve -d```).  

**Proposed commit message: (wrap lines at 72 characters)**

Implement Scroll-to-Top button on every rendered page
